sdk/core/src/main/java/com/google/cloud/trace/SpanContextFactoryTracer.java

```diff
@@ -109,15 +109,22 @@ public void setStackTrace(SpanContext context, StackTrace stackTrace) {
   }
 
   private SpanContext startSpanOptions(SpanContext parentContext, String name, Timestamp timestamp,
-      SpanKind spanKind, TraceOptions traceOptions) {
+      SpanKind spanKind, Boolean enableTrace, Boolean enableStackTrace) {
     if (timestamp == null) {
       timestamp = timestampFactory.now();
     }
     if (spanKind == null) {
       spanKind = SpanKind.UNSPECIFIED;
     }
-    if (traceOptions != null) {
-      parentContext = parentContext.overrideOptions(traceOptions);
+    if (enableTrace != null || enableStackTrace != null) {
+      TraceOptions options = parentContext.getTraceOptions();
+      if (enableTrace) {
+        options = options.overrideTraceEnabled(enableTrace);
+      }
+      if (enableStackTrace) {
```

Comment:
Should we also allow overriding to false? Making this:
```java 
if (enableStackTrace != null)
```

The same thing applies to enableTrace, but that one seems less straightforward. Does it make sense to have enableTrace=true on a parent span and enableTrace=false on a child span?